kedro-telemetry/kedro_telemetry/plugin.py

```diff
@@ -219,15 +226,10 @@ def after_context_created(self, context):
 
     @hook_impl
     def after_catalog_created(self, catalog):
+        # The user notification message is sent only once per command during the before_command_run hook
```

Comment:
What about runs that does not go through CLI, i.e. `session.run`?

Reply:
Thank you, @noklam, good point. As I understand it, in the current PR, telemetry will be sent in the `after_catalog_created()` hook without user notification. I propose addressing this issue here: [GitHub Issue #730](https://github.com/kedro-org/kedro-plugins/issues/730).

In that PR, we can modify the logic to send one heap event per user command and update user notifications accordingly:

- If it's a CLI command and the catalog is created, send one heap event in the `after_catalog_created()` hook along with a user notification message.
- If it's a CLI command and the catalog is not created, send one heap event in the `before_command_run()` hook along with a user notification message.
- If it's not a CLI command and the catalog is created, send one heap event in the `after_catalog_created()` hook along with a user notification message.

Reply:
Agree, let's address this in #730